rpc/api/eth_args.go

```diff
@@ -931,6 +933,12 @@ func (tx *tx) UnmarshalJSON(b []byte) (err error) {
 		contractCreation = true
 	)
 
+	if val, found := fields["Hash"]; found {
```

Comment:
Does not setting the hash from API input open up for errors, e.g. caller sending the wrong hash and it would be set as the tx hash. Seems better if we calculate it for the caller, or verifying the hash they provide.